src/Std/Sat/Basic.lean

```diff
@@ -0,0 +1,141 @@
+/-
+Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Josh Clune, Henrik Böving
+-/
+prelude
+import Init.NotationExtra
+
+namespace Std
+namespace Sat
+
+/--
+For variables of type `α` and formulas of type `β`, `HSat.eval a f` is meant to determine whether
+a formula `f` is true under assignment `a`.
+-/
+class HSat (α : Type u) (β : Type v) :=
```

Comment:
I did consider this as well but that's not possible. Later in LeanSAT there is an instance:
```
instance {n : Nat} : HSat (PosFin n) (Array Assignment) where
  eval := fun p arr => ∀ i : PosFin n, ¬(hasAssignment (¬p i) arr[i.1]!)
```
in the LRAT checker. This instance will be rejected because `n` cannot be inferred from the non out param `Array Assignment`.